keel-sql/src/main/resources/db/changelog/20201116-foreign-keys.yml

```diff
@@ -0,0 +1,213 @@
+databaseChangeLog:
+- changeSet:
+    id: foreign-keys-on-artifact_last_checked
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from artifact_last_checked where not exists (select 1 from delivery_artifact where artifact_last_checked.artifact_uid = delivery_artifact.uid)
+    - addForeignKeyConstraint:
+        baseTableName: artifact_last_checked
+        baseColumnNames: artifact_uid
+        constraintName: fk_artifact_last_checked_delivery_artifact
+        referencedTableName: delivery_artifact
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-current_constraint
+    author: fletch
+    changes:
+    - addForeignKeyConstraint:
+        baseTableName: current_constraint
+        baseColumnNames: environment_uid
+        constraintName: fk_current_constraint_environment
+        referencedTableName: environment
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+    - addForeignKeyConstraint:
+        baseTableName: current_constraint
+        baseColumnNames: constraint_uid
+        constraintName: fk_current_constraint_environment_artifact_constraint
+        referencedTableName: environment_artifact_constraint
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-delivery_config_artifact
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from delivery_config_artifact where not exists (select 1 from delivery_config where delivery_config_artifact.delivery_config_uid = delivery_config.uid)
```

Comment:
Ha! First time I noticed we even had this table! We should be able to get rid of it when we replace the delivery config name in `delivery_artifact` with the uid as an FK.

Reply:
Yes. I don't think this table is strictly necessary since it's not a many-to-many relationship